src/app/course-collection/course-collection.component.ts

```diff
@@ -1,15 +1,33 @@
 import { Component } from '@angular/core';
 import { Course } from 'app-models';
+import { CourseService } from 'app-services';
 
 @Component({
   selector: 'app-course-collection',
   templateUrl: './course-collection.component.html',
   styleUrls: ['./course-collection.component.scss']
 })
 export class CourseCollectionComponent {
+  private readonly courseService: CourseService;
+
   public courses: Course[];
 
-  constructor() {
-    this.courses = [new Course(), new Course(), new Course(), new Course()];
+  constructor(courseService: CourseService) {
```

Comment:
It would be better to turn off **no-parameter-properties** rule and use parameter properties 
 
```suggestion
  constructor(private readonly courseService: CourseService) {
```